fix: Return appropriate data type for time `mean` and `median`

mcrumiller opened the pull request:
Resolves #14470. Part of overall #13599.

This also moves one step closer to deprecating `Series.dt.mean` and `Series.dt.median`, as the base Series' mean and median now support **time** dtypes.

Using the example from the issue, all cases now properly output a duration:

```
06:00:00
02:00:00
06:00:00
02:00:00
shape: (1, 2)
┌──────────┬──────────┐
│ mean     ┆ median   │
│ ---      ┆ ---      │
│ time     ┆ time     │
╞══════════╪══════════╡
│ 06:00:00 ┆ 02:00:00 │
└──────────┴──────────┘
shape: (1, 2)
┌──────────┬──────────┐
│ mean     ┆ median   │
│ ---      ┆ ---      │
│ time     ┆ time     │
╞══════════╪══════════╡
│ 06:00:00 ┆ 02:00:00 │
└──────────┴──────────┘
```

MarcoGorelli commented on py-polars/tests/unit/namespaces/test_datetime.py:
```diff
@@ -956,6 +959,9 @@ def test_weekday(time_unit: TimeUnit) -> None:
         "single_dur",
         "spread_even_dur",
         "spread_skewed_dur",
+        "single_time",
+        "spread_even_time",
+        "spread_skewed_time",
```
the
```
    if s.dtype == pl.Datetime:
        assert s.median() == expected_median
```
you have on line 973, and that now be extended to dtypes other than `Datetime`? Is it only `pl.Date` that's missing?

MarcoGorelli commented on py-polars/tests/unit/namespaces/test_datetime.py:
```diff
@@ -1001,6 +1010,9 @@ def test_median(
         "single_duration",
         "spread_even_duration",
         "spread_skewed_duration",
+        "single_time",
+        "spread_even_time",
+        "spread_skewed_time",
```
same

MarcoGorelli reviewed (commented):
looks good, just some minor comments

mcrumiller commented on py-polars/tests/unit/namespaces/test_datetime.py:
```diff
@@ -1032,6 +1044,27 @@ def test_datetime_mean_with_tu(values: list[datetime], expected_mean: datetime)
+def test_datetime_median_with_tu(
+    values: list[datetime], expected_median: datetime
+) -> None:
+    assert pl.Series(values, dtype=pl.Duration("ms")).median() == expected_median
+    assert pl.Series(values, dtype=pl.Duration("ms")).dt.median() == expected_median
+    assert pl.Series(values, dtype=pl.Duration("us")).median() == expected_median
+    assert pl.Series(values, dtype=pl.Duration("us")).dt.median() == expected_median
+    assert pl.Series(values, dtype=pl.Duration("ns")).median() == expected_median
+    assert pl.Series(values, dtype=pl.Duration("ns")).dt.median() == expected_median
```
Yep. Thanks.

mcrumiller replied:
I've also cleaned up some of the IDs and added a `None` tests to the parametrization of a few.

mcrumiller commented:
@MarcoGorelli I believe I've resolved all of your issues with minor test improvements:

* Parametrized time unit where applicable
* Added a few `None` cases to existing parametrizations
* Clarified the `id` of certain parametrizations
* Added an additional aggregate median test to match the existing aggregate mean test.

MarcoGorelli commented on py-polars/tests/unit/namespaces/test_datetime.py:
```diff
@@ -1009,62 +1021,91 @@ def test_mean(
-    assert pl.Series(values, dtype=pl.Duration("ns")).dt.mean() == expected_mean
+@pytest.mark.parametrize("time_unit", ["ms", "us", "ns"])
+def test_datetime_mean_with_tu(
+    values: list[datetime], expected_mean: datetime, time_unit: str
+) -> None:
+    assert pl.Series(values, dtype=pl.Duration(time_unit)).mean() == expected_mean  # type:ignore[arg-type]
+    assert pl.Series(values, dtype=pl.Duration(time_unit)).dt.mean() == expected_mean  # type:ignore[arg-type]
```
I think you can use `time:unit: TimeUnit`, and then you don't need the type: ignores

mcrumiller replied:
Much better, thanks!

MarcoGorelli commented on py-polars/tests/unit/namespaces/test_datetime.py:
```diff
@@ -1009,62 +1021,91 @@ def test_mean(
+)
+@pytest.mark.parametrize("time_unit", ["ms", "us", "ns"])
+def test_datetime_median_with_tu(
+    values: list[datetime], expected_median: datetime, time_unit: str
```
same

MarcoGorelli commented on py-polars/tests/unit/namespaces/test_datetime.py:
```diff
@@ -1009,62 +1021,91 @@ def test_mean(
+@pytest.mark.parametrize("time_unit", ["ms", "us", "ns"])
 def test_duration_mean_with_tu(
-    values: list[timedelta], expected_mean: timedelta
+    values: list[timedelta], expected_mean: timedelta, time_unit: str
```
same

MarcoGorelli commented on py-polars/tests/unit/namespaces/test_datetime.py:
```diff
@@ -1009,62 +1021,91 @@ def test_mean(
+@pytest.mark.parametrize("time_unit", ["ms", "us", "ns"])
 def test_duration_median_with_tu(
-    values: list[timedelta], expected_median: timedelta
+    values: list[timedelta], expected_median: timedelta, time_unit: str
```
same

MarcoGorelli reviewed (approved):
thanks @mcrumiller !

mcrumiller commented:
Hi @stinodego, this one is relatively straightforward (virtually identical to #14376) and has been sitting in the queue for a while. Is there any chance you could take a look at this, so that I can get the `pl.Date` mean/median reworked with the latest main?
